### PR TITLE
Campaign list - column's sorting

### PR DESCRIPTION
**Pull request by Calanthe**
This PR introduces column's sorting on the campaign list.

screen shot 2016-11-28 at 17 33 57

@clloyd @steppenwells

### CODE REVIEW

File: public/components/CampaignList/CampaignList.js [hunk 1]
```diff
@@ -11,6 +11,40 @@ class CampaignList extends React.Component {
...
+    sortOrder: {
+      'name': true, //false = DESC, true = ASC
+      'type': false,
+      'status': false,
+      'actualValue': false,
+      'startDate': false,
+      'endDate': false
+    },
+    headersCssClasses: {
+      'name': 'campaign-list__header-order--desc name',
```
Comment (Calanthe) [inline public/components/CampaignList/CampaignList.js]: I know that you prefer to have one css class per tag @clloyd. Do you think that's an issue here?

Comment (clloyd) [reply public/components/CampaignList/CampaignList.js]: Does any of this need to be stored in the component state? Can it be implied from the redux global state? Same from the classes, can they be constructed on each render, then you don't need to keep them all in sync?

File: public/components/CampaignList/CampaignList.js [hunk 2]
```diff
@@ -11,6 +11,40 @@ class CampaignList extends React.Component {
...
+  };
+
+  setCampaignSort = (c) => {
+    const order = this.state.sortOrder[c] ? 'asc' : 'desc';
+
+    //sort campaign list
+    this.props.uiActions.setCampaignSort(c, this.state.sortOrder[c]);
+    //reset all of the headers' css classes so by default there is always up and down arrow
+    for (let value in this.state.headersCssClasses) {
+      this.state.headersCssClasses[value] = 'campaign-list__header-order ' + value;
```
Comment (clloyd) [inline public/components/CampaignList/CampaignList.js]: I don't think you need any of this information in the component state... but even if you do, don't modify this.state directly it won't trigger a render, you should call this.setState

File: public/components/CampaignList/CampaignList.js (hunk 1 above)
Comment (clloyd) [reply public/components/CampaignList/CampaignList.js]: I've no huge issue with two class names where it makes sense, I think it does in this case.

**Review by clloyd — CHANGES_REQUESTED** [review]

**Review by clloyd — APPROVED** [review]

**Pull request merged**